spec/unit/lib/temporal/worker_spec.rb

```diff
@@ -180,7 +180,23 @@ class TestWorkerActivity < Temporal::Activity
       worker.start
 
       expect(activity_poller).to have_received(:start)
+    end
 
+    it 'can have an activity poller which throttles tasks per second' do
+      activity_poller = instance_double(Temporal::Activity::Poller, start: nil)
+      expect(Temporal::Activity::Poller)
+        .to receive(:new)
+        .with('default-namespace', 'default-task-queue', an_instance_of(Temporal::ExecutableLookup), [], {max_tasks_per_second: 15, thread_pool_size: 20})
+        .and_return(activity_poller)
+
+      worker = Temporal::Worker.new(activity_max_tasks_per_second: 15)
+      allow(worker).to receive(:shutting_down?).and_return(true)
+      worker.register_workflow(TestWorkerWorkflow)
+      worker.register_activity(TestWorkerActivity)
+
+      worker.start
+
+      expect(activity_poller).to have_received(:start)
```

Comment:
nit: I'm not sure this check is needed in this spec, we're only checking that the Poller is initialised with the new option